### ISSUE TITLE
Feature Densities Always Zero

### ISSUE THREAD

**Issue by JacobOaks** [omegafragger/DDU#2]
Hello,

I'm having an issue with the active learning script. I'm running:

`CUDA_VISIBLE_DEVICES=7 python active_learning_script.py --seed 1 --model resnet18 -sn -mod --al-type gmm`

and I have set a breakpoint right before acquisition in `active_learning_script.py` here:

<img width="794" alt="Screen Shot 2022-02-04 at 1 12 13 PM" src="https://user-images.githubusercontent.com/11602410/152581071-e9c9b02f-09cf-4008-ae94-8d14ec070aa0.png">

whenever I inspect the result of `compute_density(logits, class_prob)`, density for all instances is always zero like so:

<img width="566" alt="Screen Shot 2022-02-04 at 1 18 38 PM" src="https://user-images.githubusercontent.com/11602410/152581930-304b1e98-9578-4544-af2c-0218247125e5.png">

Upon digging deeper, I noticed that the feature Gaussians always assign extremely low probability to every class, leading to all of these zeros. This happens at every acquisition step for me, regardless of the dataset size and seed (I've tried 2 seeds). I've even tried programmatically setting up a check to tell me if the densities are ever not zero for entire training runs, and it never happens. Thus, for me, the following call to `torch.topk` amounts to random selection at every acquisition step. I'm wondering if this is an issue that you've experienced.

Thanks,
Jacob

**Comment by JacobOaks** [comment]
Hello,

Just checking if you've had a chance to look at this given that it has potentially quite big implications in terms of the validity of the associated paper.

Jacob

**Comment by LinyeLi60** [comment]
I have met the same problem!
